### PR TITLE
Math Magicians: tests

### PR DESCRIPTION
**Pull request by tumainimaganiko**
In this project, we did the following :

- Set up React Testing Library.
- Write unit tests for the files operate.js and calculate.js using Jest.
- Create unit tests for all React components:
       1. Use Jest snapshots to test the components.
       2. Use React Testing Library to simulate user interaction.
- Run all our full tests suite using Jest and all tests are passing.

### CODE REVIEW

File: src/test/operate.test.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  });
+  test('operate', () => {
+    const data = operate('8', '4', 'x');
+    expect(data).toBe('32');
+  });
+  test('operate', () => {
+    const data = operate('8', '4', '÷');
+    expect(data).toBe('2');
+  });
+});
```
Comment (Munsa1) [inline src/test/operate.test.js]: - [ ] Good job adding tests for `operate.js` logic, however, we have some tests that have not been covered. Kindly ensure you include tests to check:

- That a number cannot be divided by o
image

- That the correct value is returned for the `%` modulus operation
e.g
`5%3 = 2`

- That you cannot find the `%` modulo of `0` 
image

- That unknown operations cannot be performed
image

File: src/test/calculate.test.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    total: '2',
+    next: '2',
+    operation: '+',
+  };
+  render(<Span obj={obj} setObj={() => obj} text="=" color="white" />);
+  const equal = calculate(obj, '=');
+  const button = screen.getByRole('button');
+  fireEvent.click(button);
+  expect(equal.total).toBe('4');
+});
```
Comment (Munsa1) [inline src/test/calculate.test.js]: - [ ] Again here the requirement is that we have to test for all scenarios of the `calculate.js` component, These scenarios include:

- What happens when a user clicks the `AC` button to reset the calculator
image

- What happens if there is an operation
image

- What happens if there is no operation
image

- What happens when a user clicks the  `.` point operand
image

- What happes when a user clicks the `-/+` minus/plus sign
image

- What happens when a user pressed an operation after pressing the `=` sign
image

- What happens when a user pressed an operation button and there is an existing operation
image

-

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @tumainimaganiko @otmaneechchafyky , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
Use of React Testing Library :heavy_check_mark: 
Use Of Jest Snapshots For Component Testing :heavy_check_mark: 
Descriptive Pr :heavy_check_mark: 
Equal Contribution Of Work Via Commits :heavy_check_mark: 

## Required Changes ♻️
_Check the comments under the review._
- [ ] Hey @otmaneechchafyky from here on, kindly remeber to use the imperative tense when writing your commit messages

- [ ] Kindly add the terminal command for running tests to the ReadMe file under the `Getting Started` section for a more descriptive ReadMe file
image


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- [ ] It would be nice to add your partner as a co-author to the `Author` section of your ReadMe file.
image


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TadesseDev — APPROVED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy (1)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅

## Optional suggestions

- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**